mediagrains/patterngenerators/audio/tone.py

```diff
@@ -0,0 +1,222 @@
+                                         CogAudioFormat.S24_PLANES,
+                                         CogAudioFormat.S32_PLANES,
+                                         CogAudioFormat.FLOAT_PLANES,
+                                         CogAudioFormat.DOUBLE_PLANES]:
+                planes = True
+            elif self.cog_audio_format in [CogAudioFormat.S16_PAIRS,
+                                           CogAudioFormat.S24_PAIRS,
+                                           CogAudioFormat.S32_PAIRS,
+                                           CogAudioFormat.FLOAT_PAIRS,
+                                           CogAudioFormat.DOUBLE_PAIRS]:
+                pairs = True
+            elif self.cog_audio_format in [CogAudioFormat.S16_INTERLEAVED,
+                                           CogAudioFormat.S24_INTERLEAVED,
+                                           CogAudioFormat.S32_INTERLEAVED,
+                                           CogAudioFormat.FLOAT_INTERLEAVED,
+                                           CogAudioFormat.DOUBLE_INTERLEAVED]:
+                interleaved = True
+
+            line = [formatted_sample_data[n % len(formatted_sample_data)] for n in range(offs, offs + self.samples)]
+            if planes:
+                line = line * self.channels
+            elif pairs:
+                line = [x for x in line for _ in range(0, 2)] * (self.channels//2)
+            elif interleaved:
+                line = [x for x in line for _ in range(0, self.channels)]
+
+            if depth == 16:
+                self.data_samples[offs] = struct.pack('@' + ('h'*self.samples*self.channels), *line)
+            elif depth == 24:
+                self.data_samples[offs] = b''.join(struct.pack('@i', x)[:3] for x in line)
+            elif depth == 32:
+                self.data_samples[offs] = struct.pack('@' + ('i'*self.samples*self.channels), *line)
+            elif depth == 'f':
+                self.data_samples[offs] = struct.pack('@' + ('f'*self.samples*self.channels), *line)
+            elif depth == 'd':
+                self.data_samples[offs] = struct.pack('@' + ('d'*self.samples*self.channels), *line)
+
+        return self.data_samples[offs]
+
+    def get(self, key: TimeValue, default: Optional[AUDIOGRAIN] = None) -> Optional[AUDIOGRAIN]:
+        tv = TimeValue(key, rate=Fraction(self.sample_rate))
+        sample_count = tv.as_count()
+
+        if sample_count % self.samples != 0:
+            return None
+
+        offs = sample_count % self.looplen
+
+        ag = AudioGrain(
+            self.src_id,
+            self.flow_id,
+            origin_timestamp=tv.as_timestamp(),
+            cog_audio_format=self.cog_audio_format,
+            samples=self.samples,
+            channels=self.channels,
+            rate=self.rate,
+            sample_rate=self.sample_rate)
+
+        ag.data = bytearray(self._get_samples(offs)[:ag.expected_length])
+
+        return ag
+
+
+def Tone1K(
```

Comment:
Should this be dropped for simplicity? I think it was created as a stop gap before the more general class.

Reply:
I think it hurts nothing to keep it.